src/cargo/core/compiler/build_context/target_info.rs

```diff
@@ -103,10 +103,13 @@ impl FileType {
     /// The filename for this FileType that Cargo should use when "uplifting"
     /// it to the destination directory.
     pub fn uplift_filename(&self, target: &Target) -> String {
+        // for binary crate type, `should_replace_hyphens` will always be false
         let name = if self.should_replace_hyphens {
             target.crate_name()
         } else {
-            target.name().to_string()
+            target
+                .get_binary_name()
+                .unwrap_or_else(|| target.name().to_string())
         };
         format!("{}{}{}", self.prefix, name, self.suffix)
```

Comment:
This looks like it's automatically appending a prefix/suffix such as `lib`, `.a`, and `.exe` I think. If that's the case I'm not sure that `filename` is the right name for the key in the manifest since it's only influencing part of the filename, not the whole filename.

Reply:
what about something like `binary_name` instead of `filename`? would that be more suitable?